flashlight/fl/tensor/backend/jit/opt/backends/onednn/OneDnnOpFusion.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <memory>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "flashlight/fl/tensor/backend/jit/ir/Node.h"
+#include "flashlight/fl/tensor/backend/jit/opt/Pass.h"
+
+namespace fl {
+
+/**
+ * Levearge OneDNN's post-ops to fuse operations.
+ *
+ * NOTE
+ * 1. due to OneDNN limitation, binary post-op only supports rhs argument.
+ * 2. currently we avoid recomputation -- fuse iff intermediate nodes are _only_
+ *    used as input nodes in the chain. There might be places where benefit of
+ *    aggressive fusion outweighs cost of recomputation, need to investigate
+ *    more (think Halide).
+ *
+ * n1   n2
+ *  \  /
+ *   b2   n3
+ *    \  /
+ *     b1
+ *
+ * -->
+ *
+ *    n1 n2 n3
+ *     \ |  /
+ * ---------------- CustomNode with evaluation logic that uses OneDNN post-op
+ * | n1 = n1 + n2 |
+ * | n1 = n2 + n3 |
+ * ----------------
+ *
+ * TODO
+ * - leverage commutativity of certain binops to bypass the rhs-only limitation
+ *   of OneDNN binary post-ops.
+ * - support more than just binop fusion
+ */
+class OneDnnOpFusion : public Pass {
```

Comment:
`OneDnnOpFusion` is quite generic -- is the idea here that this spans a large set of optimizations beyond binop fusion?

Reply:
That's the intention indeed.

The thing with OneDNN's post-ops is that, we can append different ops in the chain (eltwise, sum, and even convolution). So ideally we don't wanna restrict the fusion to be "binops only".

I'll look more into how to cleanly do this after I throw in other IR nodes (eltwise, etc.)